firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java

```diff
@@ -694,7 +694,9 @@ private void trackLimboChange(LimboDocumentChange change) {
   private void pumpEnqueuedLimboResolutions() {
     while (!enqueuedLimboResolutions.isEmpty()
         && activeLimboTargetsByKey.size() < maxConcurrentLimboResolutions) {
-      DocumentKey key = enqueuedLimboResolutions.remove();
+      Iterator<DocumentKey> it = enqueuedLimboResolutions.iterator();
```

Comment:
Looks like you need to import `Iterator`.

Reply:
Done.